python/graphstorm/gconstruct/utils.py

```diff
@@ -246,6 +250,8 @@ def multiprocessing_data_read(in_files, num_processes, user_parser):
         return_dict = {}
         while len(return_dict) < num_files:
             file_idx, vals= res_queue.get()
+            if vals is None:
+                raise RuntimeError("One of the worker processes fails. Stop data processing.")
```

Comment:
It seems file_idx set in L188 is not used.
BTW, why not raise the RuntimeError in L188?

Reply:
the main idea is that we need to capture all exceptions in the worker processes and stop the master process whenever an error happens in the worker process.